src/dotnet/commands/dotnet-buildserver/shutdown/BuildServerShutdownCommand.cs

```diff
@@ -0,0 +1,94 @@
+        private readonly IReporter _reporter;
+        private readonly IReporter _errorReporter;
+
+        public BuildServerShutdownCommand(
+            AppliedOption options,
+            ParseResult result,
+            IBuildServerManager msbuildServerManager = null,
+            IBuildServerManager vbcscompilerServerManager = null,
+            IBuildServerManager razorServerManager = null,
+            IReporter reporter = null)
+            : base(result)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _msbuildServerManager = msbuildServerManager ?? new MSBuildServerManager();
+            _vbcscompilerServerManager = vbcscompilerServerManager ?? new VBCSCompilerServerManager();
+            _razorServerManager = razorServerManager ?? new RazorServerManager();
+            _reporter = reporter ?? Reporter.Output;
+            _errorReporter = reporter ?? Reporter.Error;
+        }
+
+        public override int Execute()
+        {
+            bool msbuild = _options.ValueOrDefault<bool>("msbuild");
+            bool vbcscompiler  = _options.ValueOrDefault<bool>("vbcscompiler");
+            bool razor = _options.ValueOrDefault<bool>("razor");
+            bool all = !msbuild && !vbcscompiler && !razor;
+            bool success = true;
+
+            if (msbuild || all)
+            {
+                success &= Shutdown(_msbuildServerManager, LocalizableStrings.MSBuildServer);
+            }
+
+            if (vbcscompiler || all)
+            {
+                success &= Shutdown(_vbcscompilerServerManager, LocalizableStrings.VBCSCompilerServer);
+            }
+
+            if (razor || all)
+            {
+                success &= Shutdown(_razorServerManager, LocalizableStrings.RazorServer);
+            }
+
+            return success ? 0 : 1;
+        }
+
+        private bool Shutdown(IBuildServerManager manager, string serverName)
+        {
+            _reporter.Write(string.Format(LocalizableStrings.ShuttingDownServer, serverName));
+
+            var result = manager.Shutdown();
+
+            switch (result.Kind)
+            {
+                case ResultKind.Success:
+                    _reporter.WriteLine(LocalizableStrings.ShutDownSucceeded.Green());
+                    return true;
+
+                case ResultKind.Skipped:
+                    _reporter.WriteLine(string.Format(LocalizableStrings.ShutDownSkipped, result.Message).Cyan());
+                    return true;
+
+                case ResultKind.Failure:
+                    _errorReporter.WriteLine(string.Format(LocalizableStrings.ShutDownFailed, result.Message).Red());
+
+                    if (Reporter.IsVerbose)
+                    {
+                        Reporter.Verbose.WriteLine(result.Exception.ToString().Red());
+                    }
+                    return false;
+
+                default:
+                    throw new InvalidOperationException();
```

Comment:
>throw new InvalidOperationException(); [](start = 20, length = 38)

Don't think this is the right exception. The operation isn't invalid the operation returned an illegal enum value. Should consider adding the value of the enum and name of the enum type to the error so it's diagnosable.